common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -786,7 +787,7 @@ def choose_new_seed(self):
             self.seed = 1
         elif self.rerandomize == RANDOMIZATION.PER_STUDENT and hasattr(self.runtime, 'seed'):
             # see comment on randomization_bin
-            self.seed = randomization_bin(self.runtime.seed, str(self.location).encode('utf-8'))
+            self.seed = randomization_bin(self.seed, str(self.location).encode('utf-8'))
```

Comment:
Before the `seed` field is set it is going to be `None`. Since `runtime.seed == user.id if user is not None else 0` we can put this logic here using the user id value from the user service.